pkg/segment/aggregations/generateevents.go

```diff
@@ -86,3 +99,112 @@ func PerformGenTimes(aggs *structs.QueryAggregators) error {
 
 	return nil
 }
+
+func checkCSVFormat(filename string) bool {
+	return strings.HasSuffix(filename, ".csv") || strings.HasSuffix(filename, ".csv.gz")
+}
+
+// Ensure record and column lengths are valid before calling this function.
+func createRecord(columnNames []string, record []string) map[string]interface{} {
+	recordMap := make(map[string]interface{})
+	for i, col := range columnNames {
+		recordMap[col] = record[i]
+	}
+	return recordMap
+}
+
+func PerformInputLookup(aggs *structs.QueryAggregators) error {
+	var reader *csv.Reader
+	if aggs.GenerateEvent.InputLookup == nil {
+		return fmt.Errorf("PerformInputLookup: InputLookup is nil")
+	}
+	filename := aggs.GenerateEvent.InputLookup.Filename
+
+	if !checkCSVFormat(filename) {
+		return fmt.Errorf("PerformInputLookup: Only CSV format is currently supported")
+	}
+
+	workingDir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("PerformInputLookup: Error while getting current working directory, err: %v", err)
+	}
+	filepath := workingDir + "/" + config.GetLookupPath() + filename
+
+	file, err := os.Open(filepath)
```

Comment:
Why do you have to get the working directory? Without the working directory you should get a relative path, and you can `os.Open()` that.

Also when you do `config.GetLookupPath() + filename`, you're assuming the lookup path ends with a `/`. You should use https://pkg.go.dev/path/filepath#Join instead, as that handles both cases

Reply:
addressed in the latest commit.